src/common/utils.h

```diff
@@ -327,17 +349,7 @@ inline std::string dispatch_mode_string(const DispatchMode x) {
 
 
 /*! \brief get string representation of storage_type */
-inline std::string stype_string(const int x) {
-  switch (x) {
-    case kDefaultStorage:
-      return "default";
-    case kCSRStorage:
-      return "csr";
-    case kRowSparseStorage:
-      return "row_sparse";
-  }
-  return "unknown";
-}
+std::string stype_string(const int x);
```

Comment:
why not inline?

Reply:
there isn't a big reason for this. I didn't want to compile everything whenever I change a header file, so I moved the definition of the function in the .cc file. I'll move the code back.